### PR TITLE
Remove delayed commits option

### PR DESCRIPTION
**Pull request by nickva**
## Overview

Delayed commits option is removed

This effectively removes a lot `couch_db:ensure_full_commit/1,2` calls.

However`/ _ensure_full_commit` HTTP API was left as is since replicator from older versions of CouchDB would call that, it just returns the start time as if ensure_commit function was called.

Issue: https://github.com/apache/couchdb/issues/2165

## Testing recommendations

```
make eunit
make elixir
```

## Related Issues or Pull Requests

Issue: https://github.com/apache/couchdb/issues/2165

## Checklist

- [x] Code is written and works correctly
- [x] Changes are covered by tests
- [x] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [x] A PR for documentation changes will be been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

File: src/couch/src/couch_db_int.hrl [hunk 1]
```diff
@@ -32,6 +32,8 @@
     before_doc_update = nil, % nil | fun(Doc, Db) -> NewDoc
     after_doc_read = nil,    % nil | fun(Doc, Db) -> NewDoc
 
+    % feature removed in 3.x, but field kept to avoid changing db record size
+    % and breaking rolling cluster upgrade
     waiting_delayed_commit = nil,
```
Comment (davisp) [inline src/couch/src/couch_db_int.hrl]: You might rename this field which will cause the compiler to complain if its referneced anywhere that was missed. Something simple like `waiting_delayed_commit_deprecated` or similar.

Comment (nickva) [reply src/couch/src/couch_db_int.hrl]: Good call

**Comment by davisp** [comment]
On that last bit. delayed_commits would never lead to a corrupted database as you'd only be missing the last second or less of writes. But if people start looking for non delayed_writes settings and find the fsync_options then we're in Corruption Land. And nobody likes Corruption Land. Its the worst of the worst theme parks.

**Comment by davisp** [comment]
Also for a user that finds the `fsync_options` I could totally understand someone not very well versed in write barriers thinking "Well, I only really need an fsync after the header..." which seems reasonable until you learn about reordered writes which could leave a valid header pointing at garbage.

**Comment by nickva** [comment]
Documentation PR: https://github.com/apache/couchdb-documentation/pull/445

**Comment by PiotrKozimor** [comment]
I currently evaluate CouchDB for use with RPi CM3+ and the bigest issue is eMMC memory bandwidth that couchDB is generating because of fsync calls. @davisp can I somehow recover `delayed_commits`  feature? What is the core issue with `fsync_options`?

> On that last bit. delayed_commits would never lead to a corrupted database as you'd only be missing the last second or less of writes. But if people start looking for non delayed_writes settings and find the fsync_options then we're in Corruption Land. And nobody likes Corruption Land. Its the worst of the worst theme parks.

**Comment by davisp** [comment]
@PiotrKozimor The only way to recover the `delayed_commits` option would be to run an older 2.x release. As to `fsync_options` leaving that would have allowed users to disable fsyncs in an unsafe manner that would lead to corruption of a database (whereas `delayed_commits` might lose the last second of data, removing fsyncs can actually result in a database that is unreadable by CouchDB).

Another thing to keep in mind is that the next version of CouchDB will be based on FoundationDB which will change a lot of this considerably as the storage implementation will be different. There's interest in implementing the ability for CouchDB to run against a storage engine that would theoretically work much better on embedded devices, although this is not currently a promised feature. If you're interested in that work and/or contributing to that you should sign up for the dev@couchdb.apache.org mailing list.

**Pull request merged**